File: android/app/src/main/java/app/ezber/android/models/Surah.kt
package app.ezber.android.models

/** One of the 114 chapters of the Quran (content DB table `surahs`). */
data class Surah(
    val id: Int,
    val nameArabic: String,
    val nameLatin: String,
    val nameEnglish: String,
    val verseCount: Int,
    val revelationPlace: RevelationPlace,
    val bismillahPre: Boolean = true,
) {
    val subtitle: String get() = "$nameEnglish · $verseCount verses"
}

enum class RevelationPlace(val label: String) {
    MECCAN("Meccan"),
    MEDINAN("Medinan"),
    ;

    companion object {
        /** Maps content_schema.sql`s `revelation` ('Meccan' | 'Medinan'). */
        fun fromSchema(value: String?): RevelationPlace =
            if (value.equals("Medinan", ignoreCase = true)) MEDINAN else MECCAN
    }
}

File: ios/Ezber/Models/Surah.swift
import Foundation

/// One of the 114 chapters of the Quran.
struct Surah: Identifiable, Hashable, Codable {
    let id: Int
    let nameArabic: String
    let nameLatin: String
    let nameEnglish: String
    let verseCount: Int
    let revelationPlace: RevelationPlace

    var subtitle: String {
        "\(nameEnglish) · \(verseCount) verses"
    }
}

enum RevelationPlace: String, Codable, CaseIterable {
    case meccan
    case medinan

    var label: String {
        switch self {
        case .meccan: return "Meccan"
        case .medinan: return "Medinan"
        }
    }
}
